Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/common/azure/serverless/AzureSparkServerlessCluster.java

```diff
@@ -517,6 +533,63 @@ public AzureHttpObservable getHttp() {
                 .defaultIfEmpty(this);
     }
 
+    @Nullable
+    private UpdateSparkResourcePool preparePatchResourcePool(int workerTargetInstanceCount) {
+        if (master == null || worker == null) {
+            return null;
+        }
+
+        UpdateSparkResourcePool patchBody = new UpdateSparkResourcePool();
+
+        return patchBody
+                .withName(getName())
+                .withProperties(new UpdateSparkResourcePoolParameters()
+                        .withSparkResourceCollection(Arrays.asList(
+                                new UpdateSparkResourcePoolItemParameters()
+                                        .withName(SparkNodeType.SPARK_WORKER)
+                                        .withTargetInstanceCount(workerTargetInstanceCount)
+                        )));
+    }
+
+    private Observable<SparkResourcePool> updateResourcePoolRequest(int workerTargetInstanceCount) {
+        URI uri = getUri();
+
+        UpdateSparkResourcePool patchBody = preparePatchResourcePool(workerTargetInstanceCount);
+        if (patchBody == null) {
+            return Observable.error(new IllegalArgumentException("Spark master and worker is not stable yet. Please retry until they are stable."));
+        }
+
+        String json = patchBody.convertToJson()
+                .orElseThrow(() -> new IllegalArgumentException("Bad Spark resource pool arguments to patch"));
+
+        StringEntity entity = new StringEntity(json, StandardCharsets.UTF_8);
+        entity.setContentType("application/json");
+
+        return getHttp()
+                .withUuidUserAgent()
+                .patch(uri.toString(), entity, null, null, SparkResourcePool.class);
+    }
+
+    public Observable<AzureSparkServerlessCluster> update(int workerTargetInstanceCount) {
+        return updateResourcePoolRequest(workerTargetInstanceCount)
+                .map(resourcePoolResp -> {
```

Comment:
Can it be replaced with `.map(this::updateWithResponse)`?

Reply:
The response here is incomplete so we can't just replace the local cluster with the response.

---
In reply to: [197368287](https://github.com/Microsoft/azure-tools-for-java/pull/1734#discussion_r197368287) [](ancestors = 197368287)